syftbox/client/benchmark/netstats_http.py

```diff
@@ -0,0 +1,151 @@
+    redirect: float
+    """Time taken for all redirection steps before the final request"""
+
+
+class HTTPPerfStats:
+    """Measure HTTP connection performance using curl_cffi"""
+
+    def __init__(self, url: str):
+        self.url = url
+        self.connect_timeout: int = 30
+        self.total_timeout: int = 60
+        self.max_redirects: int = 5
+
+    def get_stats(self, n_runs: int) -> HTTPTimingStats:
+        """Aggregate performance stats from multiple runs"""
+
+        measurements: list[HTTPTimings] = []
+        for _ in range(n_runs):
+            if stats := self.__make_request(self.url):
+                measurements.append(stats)
+            time.sleep(0.5)  # Small delay between requests
+
+        if not measurements:
+            raise RuntimeError("No successful measurements")
+
+        # Calculate aggregated stats
+        def _stats_for_measurement(metric: str) -> Stats:
+            values = [getattr(m, metric) for m in measurements]
+            return Stats.from_values(values)
+
+        return HTTPTimingStats(
+            dns=_stats_for_measurement("dns"),
+            tcp_connect=_stats_for_measurement("tcp_connect"),
+            ssl_handshake=_stats_for_measurement("ssl_handshake"),
+            send=_stats_for_measurement("send"),
+            server_wait=_stats_for_measurement("server_wait"),
+            content=_stats_for_measurement("content"),
+            total=_stats_for_measurement("total"),
+            redirect=_stats_for_measurement("redirect"),
+            success_rate=len(measurements) / n_runs * 100,
+        )
+
+    def __make_request(self, url: str) -> Optional[HTTPTimings]:
+        """Get HTTP performance stats for a single request"""
+
+        buff = BytesIO()
+        curl = Curl()
+
+        opts = {
+            CurlOpt.URL: url.encode(),
+            CurlOpt.WRITEDATA: buff,
+            CurlOpt.FOLLOWLOCATION: 1,
+            CurlOpt.MAXREDIRS: self.max_redirects,
+            CurlOpt.CONNECTTIMEOUT: self.connect_timeout,
+            CurlOpt.TIMEOUT: self.total_timeout,
+            CurlOpt.SSL_VERIFYPEER: 1,
+            CurlOpt.SSL_VERIFYHOST: 2,
+        }
+        [curl.setopt(option, value) for option, value in opts.items()]
+
+        try:
+            curl.perform()
+
+            # Curl Timings https://curl.se/libcurl/c/curl_easy_getinfo.html#TIMES
+            # from start of request to stage (in microseconds)
+            namelookup_t = curl.getinfo(CurlInfo.NAMELOOKUP_TIME)  # DNS lookup
+            connect_t = curl.getinfo(CurlInfo.CONNECT_TIME)
+            appconnect_t = curl.getinfo(CurlInfo.APPCONNECT_TIME)
+            pretransfer_t = curl.getinfo(CurlInfo.PRETRANSFER_TIME)
+            starttransfer_t = curl.getinfo(CurlInfo.STARTTRANSFER_TIME)  # TTFB
+            total_t = curl.getinfo(CurlInfo.TOTAL_TIME)  # total time or TTLB
+            redirect_t = curl.getinfo(CurlInfo.REDIRECT_TIME)
+
+            # 1. Time spent waiting for the initial response
```

Comment:
#1 has nothing beneath it while the rest of them does. This is a bit confusing

Reply:
Fixed by: [e46168c](https://github.com/OpenMined/syft/pull/463/commits/e46168c93d1d50345a7a4b47b0270c2a6fd0e6a1)